lib/livebook/file_system/s3.ex

```diff
@@ -113,8 +119,28 @@ defmodule Livebook.FileSystem.S3 do
       :access_key_id,
       :secret_access_key
     ])
+    |> put_region_from_uri()
     |> validate_required([:bucket_url, :access_key_id, :secret_access_key])
     |> Livebook.Utils.validate_url(:bucket_url)
+    |> put_id()
+  end
+
+  defp put_region_from_uri(changeset) do
+    case get_field(changeset, :bucket_url) do
+      nil -> changeset
+      bucket_url -> put_change(changeset, :region, region_from_uri(bucket_url))
```

Comment:
I think the existing file systems won't have this information, right? Is that an issue? If so, I wonder if we should just start from scratch and ignore all old type systems that exist today.

Reply:
@josevalim they have, and for the very old ones that don't we compute it when loading :)

Reply:
My concern is that we would need to keep the "compute when loading" code forever. If we can't compute on the migration, let's just nuke the data. Now is the best time to do it. :)

Reply:
The thing is that `:region` is optional as defined by `new/2`, so it's fine:

https://github.com/livebook-dev/livebook/blob/ed91c9be17637daa77feaab358499cf237681237/lib/livebook/file_system/s3.ex#L39-L43

Reply:
Ok, resolving this then. :)